src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,236 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+import duke.util.DateTimeHandler;
+
+/**
+ * Encapsulates the handling of user's input.
+ */
+public class Parser {
+
+    /** Storage for parser to write data to. */
+    private final Storage storage;
+
+    /** UI for parser to display messages. */
+    private final Ui ui;
+
+    /** Task list for parser to add/edit/delete tasks. */
+    private final TaskList taskList;
+
+    /**
+     * Creates a new parser from a storage.
+     *
+     * @param storage Storage of Duke's data.
+     */
+    public Parser(Storage storage) {
+        this.storage = storage;
+        this.ui = this.storage.getUi();
+        this.taskList = this.storage.getTaskList();
+    }
+
+    /**
+     * Processes the input fed to Duke.
+     *
+     * @param input Input string.
+     * @throws DukeException If input is invalid.
+     */
+    public void processInput(String input) throws DukeException {
```

Comment:
Perhaps can consider changing the variable input to something like userCommand.

Reply:
Changed to userInput.